src/utils/requestNavigationHandlers.ts

```diff
@@ -22,8 +22,236 @@ import { findWalletWithAccount } from '@/helpers/findWalletWithAccount';
+      const network = ethereumUtils.getNetworkFromChainId(request.account?.networkId ?? ChainId.mainnet);
+
+      // @ts-expect-error - coinbase host protocol types are NOT correct e.g. {"data": [72, 101, 108, 108, 111, 32, 119, 111, 114, 108, 100], "type": "Buffer"}
+      if ((action as PersonalSignAction).params.message && (action as PersonalSignAction).params.message.type === 'Buffer') {
+        // @ts-expect-error - coinbase host protocol types are NOT correct e.g. {"data": [72, 101, 108, 108, 111, 32, 119, 111, 114, 108, 100], "type": "Buffer"}
+        const messageFromBuffer = toUtf8String(Buffer.from((action as PersonalSignAction).params.message.data, 'hex'));
+        (action as PersonalSignAction).params.message = messageFromBuffer;
+      }
+
+      const payload = {
+        method: action.method,
+        params: constructEthereumActionPayload(action),
+      };
+
+      const displayDetails = await getRequestDisplayDetails(payload, nativeCurrency, network);
+
+      const requestWithDetails: RequestData = {
+        dappName: session?.dappName ?? session?.dappId ?? '',
+        dappUrl: session?.dappURL ?? '',
+        imageUrl: session?.dappImageURL ?? '',
+        address: (action as PersonalSignAction).params.address ?? accountAddress,
+        network,
+        payload,
+        displayDetails,
+      };
+
+      console.log(JSON.stringify(requestWithDetails, null, 2));
+
+      return new Promise((resolve, reject) => {
+        const onSuccess = async (result: string) => {
+          logger.debug(`Ethereum action approved: [${action.method}]: ${result}`);
+          const success = await approveAction(action, { value: JSON.stringify(result) });
+          resolve(success);
+        };
+
+        const onCancel = async (error?: Error) => {
+          if (error) {
+            logger.debug(`Ethereum action rejected: [${action.method}]: ${error.message}`);
+            await rejectAction(action, {
+              message: error.message,
+              code: 4001,
+            });
+            reject(error.message);
+          } else {
+            logger.debug(`Ethereum action rejected: [${action.method}]: User rejected request`);
+            await rejectAction(action, {
+              message: 'User rejected request',
+              code: 4001,
+            });
+            reject('User rejected request');
+          }
+        };
+
+        Navigation.handleAction(Routes.CONFIRM_REQUEST, {
+          transactionDetails: requestWithDetails,
+          onSuccess,
+          onCancel,
+          onCloseScreen: noop,
+          network,
+          address: accountAddress,
+          source: RequestSource.MOBILE_WALLET_PROTOCOL,
+        });
+      });
+    } else {
+      logger.error(new RainbowError(`[handleMobileWalletProtocolRequest]: Unsupported action type, ${action}`));
+      return false;
+    }
+  };
+
+  const handleActions = async (actions: typeof request.actions, currentIndex: number = 0): Promise<boolean> => {
```

Comment:
recursively loop over batched actions and process them.

Reply:
Most times this will only be processing 1 action, but nice to support batched actions if an application wants to send us action that way.